components/client-common.rst

```diff
@@ -81,3 +81,76 @@ PluginClient
 ------------
 
 See :doc:`/plugins/introduction`
+
+HttpClientPool
+--------------
+
+The ``HttpClientPool`` allows to balance requests between a pool of ``HttpClient`` and/or ``HttpAsyncClient``.
+
+The use cases are:
+
+- Using a cluster (like an Elasticsearch service with multiple master nodes)
+- Using fallback servers with the combination of the ``RetryPlugin`` (see :doc:`/plugins/retry`)
+
+You can attach HTTP clients to this kind of client by using the ``addHttpClient`` method::
+
+    use Http\Client\Common\HttpClientPool\LeastUsedClientPool;
+    use Http\Discovery\HttpAsyncClientDiscovery;
+    use Http\Discovery\HttpClientDiscovery;
+    use Http\Discovery\MessageFactoryDiscovery;
+
+    $messageFactory = MessageFactoryDiscovery::find();
+
+    $httpClient = HttpClientDiscovery::find();
+    $httpAsyncClient = HttpAsyncClientDiscovery::find();
+
+    $httpClientPool = new LeastUsedClientPool();
+    $httpClientPool->addHttpClient($httpClient);
+    $httpClientPool->addHttpClient($httpAsyncClient);
+
+    $httpClientPool->sendRequest($messageFactory->createRequest('GET', 'http://example.com/update'));
+
+Clients added to the pool are decorated with the ``HttpClientPoolItem`` class unless they already are an instance of this class.
+
+The pool item class lets the pool be aware of the number of requests currently being processed by that client.
```

Comment:
Merge this paragraph with the paragraph above. We are still talking about the pool item class.